docs/tbtc-liquidation-recovery.adoc

```diff
@@ -0,0 +1,178 @@
+toc::[]
+
+== Background
+
+.As described in xref:rfc/rfc-2.adoc[RFC 2]:
+[quote]
+____
+include::rfc/rfc-2.adoc[tag=background]
+____
+
+Keep ECDSA client implements a mechanism mentioned above to recover BTC after
+a tBTC Deposit got liquidated and split the amount among the Keep signers holding
+the BTC.
+
+== Configuration
+
+To enable the tBTC Liquidation Recovery mechanism you have to update TOML configuration
+file with details of destination (`BeneficiaryAddress`) to which Bitcoin funds should
+be sent once liquidation happens.
+
+.BeneficiaryAddress in TOML Config File
+```toml
+[Extensions.TBTC.Bitcoin]
+BeneficiaryAddress = "<your btc address or *pub key for a hierarchical deterministic wallet>"
+```
+
+The Beneficiary Address can be provided in one of two formats:
+
+1. A simple Bitcoin address, that will be used for all transactions.
+
+2. An extended public key (*pub), that will be used to derive unique addresses
+for each transaction, see <<Bitcoin Addresses Derivation>> section.
+
+For examples see xref:run-keep-ecdsa.adoc#example-beneficiary-addresses[Example Beneficiary Addresses].
+
+For all configuration parameters please see xref:run-keep-ecdsa.adoc#config-extensions-tbtc[tBTC Extension configuration properties].
+
+== Bitcoin Addresses Derivation
+
+`BeneficiaryAddress` can be provided as an extended public key described by 
+https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki[BIP 32].
+
+Type of Bitcoin address that will be calculated is resolved based on the `*pub`
+prefix, according to https://github.com/satoshilabs/slips/blob/master/slip-0132.md[SLIP-0132].
+
+.Addresses Encoding Based on *pub Prefix
+[%header,cols="^1m,^2,3m"]
+|===
+|Prefix
+|Address Encoding
+^|Sample Address
+
+|xpub
+|P2PKH (Legacy)
+|1L3mGzLD9XLXKxkgLB5Gc6rW6s4ik4yd2r
+
+|ypub
+|P2WPKH nested in P2SH (Segwit)
+|3DMLhAwVrJxePftA8Q5emyoNFCj8Jn1q64
+
+|zpub
+|P2WPKH (Native Segwit)
+|bc1q6rczu6pcv2eq9gh9xvrwx6y9uhxk88f8xm020g
+|===
+
+For more examples see xref:run-keep-ecdsa.adoc#example-beneficiary-addresses[Example Beneficiary Addresses].
+
+If the extended public key is provided the client will resolve a unique and unused
+Bitcoin address for each Bitcoin transaction.
+
+Addresses are resolved starting from `0` index. The index is incremented by 1 in
+case of the address being used already.
+
+The highest used index for the extended public keys will be stored in the client's
+local storage directory under `bitcoin/derivation_indexes/<EXTENDED_PUBLIC_KEY_ID>/<INDEX>`
+(e.g. `bitcoin/derivation_indexes/xpub_b1wex5vq/3`).
+
+Additionally to the local storage check the client will reach Bitcoin API to
+check if there are any existing transactions for the given address.
+
+If the Bitcoin connection is missing the client will check the address usage
+based on the local storage only.
+
+=== Bitcoin Chain Connectivity
+
+Connectivity to Bitcoin chain API is used for liquidation recovery handling but is not required for the process to complete.
```

Comment:
Here we call it the Bitcoin chain API, but above we cal it just the Bitcoin API. I think we can standardize one one for consistency. (My vote is Bitcoin API but my preference isn't strong)

Reply:
4d445deb03bdd43954ecc1262a61675a1d0f35f9